src/test/TestUSDCIntegration.t.sol

```diff
@@ -0,0 +1,144 @@
+
+
+        termStrategy = Strategy(address(strategy));
+
+        repoToken1WeekAuction = new MockTermAuction(repoToken1Week);
+
+        vm.startPrank(management);
+        termStrategy.setCollateralTokenParams(address(mockCollateral), 0.5e18);
+        termStrategy.setTimeToMaturityThreshold(3 weeks);
+        termStrategy.setRepoTokenConcentrationLimit(1e18);
+        termStrategy.setRequiredReserveRatio(0);
+        termStrategy.setdiscountRateMarkup(0);
+        vm.stopPrank();
+
+        // start with some initial funds
+        mockUSDC.mint(address(strategy), 100e6);
+
+        initialState.totalAssetValue = termStrategy.totalAssetValue();
+        initialState.totalLiquidBalance = termStrategy.totalLiquidBalance();
+    }
+
+    function _submitOffer(bytes32 idHash, uint256 offerAmount) private returns (bytes32) { 
+        // test: only management can submit offers
+        vm.expectRevert("!management");
+        bytes32[] memory offerIds = termStrategy.submitAuctionOffer(
+            repoToken1WeekAuction, address(repoToken1Week), idHash, bytes32("test price"), offerAmount
+        );        
+
+        vm.prank(management);
+        offerIds = termStrategy.submitAuctionOffer(
+            repoToken1WeekAuction, address(repoToken1Week), idHash, bytes32("test price"), offerAmount
+        );        
+
+        assertEq(offerIds.length, 1);
+
+        return offerIds[0];
+    }
+
+    function testSellRepoTokenSubmitOfferAndCloseAuction() public {       
+        address testUser = vm.addr(0x11111);  
+        mockUSDC.mint(testUser, 1e18);
+        repoToken1Month.mint(testUser, 1000e18);
+
+        vm.startPrank(testUser);
+        mockUSDC.approve(address(mockYearnVault), type(uint256).max);
+        mockYearnVault.deposit(1e18, testUser);
+        repoToken1Month.approve(address(strategy), type(uint256).max);
+        termStrategy.sellRepoToken(address(repoToken1Month), 1e6);
+        mockYearnVault.withdraw(1e18, testUser, testUser);
+        vm.stopPrank();
+
+        address[] memory holdings = termStrategy.repoTokenHoldings();
+
+        assertEq(holdings.length, 1);
+
+
+        bytes32 offerId1 = _submitOffer(bytes32("offer id hash 1"), 1e6);
+        bytes32[] memory offerIds = new bytes32[](1);
+        offerIds[0] = offerId1;
+        uint256[] memory fillAmounts = new uint256[](1);
+        fillAmounts[0] = 1e6;
+        uint256[] memory repoTokenAmounts = new uint256[](1);
+        repoTokenAmounts[0] = _getRepoTokenAmountGivenPurchaseTokenAmount(
+            1e6, repoToken1Week, TEST_REPO_TOKEN_RATE
+        );
+
+
+        repoToken1WeekAuction.auctionSuccess(offerIds, fillAmounts, repoTokenAmounts);
+
+        holdings = termStrategy.repoTokenHoldings();
+
+        // test: 1 holding because auctionClosed not yet called
+        assertEq(holdings.length, 1);
+
+        termStrategy.auctionClosed();
+        holdings = termStrategy.repoTokenHoldings();
+        assertEq(holdings.length, 2);
```

Comment:
can we check here if the repo tokens are properly sorted by maturity?

Reply:
good idea